view/frontend/templates/js/replacejs.phtml

```diff
@@ -1122,6 +1124,9 @@ $isLoadTrackJsDynamic = $block->isDisableTrackJsOnNonBoltPages() && $isLoadConne
                             && boltCart.data_id !== undefined
                             && data.data_id<=boltCart.data_id
                         ) {
+                            if(boltCart.status === 'failure'){
```

Comment:
Let's move this code to functuon boltCartDataListener
If we left it here it doesn't catch the case when boltCart was correct initially and some issue happened later.